### ISSUE TITLE
inconsistent default varIDs for subscribe, subscribeContext

### ISSUE THREAD

**Issue by namdre** [eclipse-sumo/sumo#15431]
the traci client has a concept of a default response for every domain (i.e. https://github.com/eclipse-sumo/sumo/blob/ce7940e5e491ea8547d6bacabbced90a711e8a96/tools/traci/_vehicle.py#L300) whereas libsumo uses [-1] as default (because an empty list would signal subscription removal) and thus only returns object IDs.

**Comment by namdre** [comment]
somewhat related to #15430

**Comment by behrisch** [comment]
I agree that the API is not 100% identical but the behavior should be correct, because the "-1" is used as a hacky workaround, see the corresponding code in Helper.cpp: https://github.com/eclipse-sumo/sumo/blob/ce7940e5e491ea8547d6bacabbced90a711e8a96/src/libsumo/Helper.cpp#L138. Also the tests run just fine with traci and libsumo. @namdre Can you give an example which does not work?

**Comment by namdre** [comment]
*traci/contextSubscriptions/matrix_test_defaults* fails for complex.libsumo and complex.libtraci

**Comment by behrisch** [comment]
This test passes `None` explicitly instead of relying on the defaults by not using the function parameter at all. This is a different use case and should now be fixed in libsumo. It is still not working in libtraci and I don't know why (yet).
